trieste/space.py

```diff
@@ -32,6 +34,186 @@
+    A wrapper class for linear constraints on variables. The constraints expression is of the form::
+
+        lb <= A @ x <= ub
+
+    :param A: The matrix defining the linear constraints with shape [M, D], where M is the
+        number of constraints.
+    :param lb: The lower bound of the constraint. Should be a scalar or of shape [M].
+    :param ub: The upper bound of the constraint. Should be a scalar or of shape [M].
+    :param keep_feasible: Keep the constraints feasible throughout optimization iterations if this
+        is `True`.
+    """
+
+    def __init__(
+        self,
+        A: TensorType,
+        lb: Sequence[float] | TensorType,
+        ub: Sequence[float] | TensorType,
+        keep_feasible: bool = False,
+    ):
+        super().__init__(A, lb, ub, keep_feasible=keep_feasible)
+
+    def residual(self, points: TensorType) -> TensorType:
+        """
+        Calculate the residuals between the constraint function and its lower/upper limits.
+
+        >>> points = tf.constant([[-1.0, 0.4], [-1.0, 0.6], [0.0, 0.4]], dtype=DEFAULT_DTYPE)
+        >>> lc = LinearConstraint(A=np.array([[-1.0, 1.0], [1.0, 0.0]]),
+        ...                       lb=np.array([-0.4, 0.5]), ub=np.array([-0.2, 0.9]))
+        >>> lc.residual(points)
+        <tf.Tensor: shape=(3, 4), dtype=float64, numpy=
+        array([[ 1.8, -1.5, -1.6,  1.9],
+               [ 2. , -1.5, -1.8,  1.9],
+               [ 0.8, -0.5, -0.6,  0.9]])>
+
+
+        :param points: The points to calculate the residuals for, with shape [..., D].
+        :return: A tensor containing the lower and upper residual values with shape [..., M*2].
+        """
+        tf.debugging.assert_rank_at_least(points, 2)
+        non_d_axes = np.ones_like(points.shape)[:-1]  # Avoid adding axes shape to static graph.
+        lb = tf.cast(tf.reshape(self.lb, (*non_d_axes, -1)), dtype=points.dtype)
+        ub = tf.cast(tf.reshape(self.ub, (*non_d_axes, -1)), dtype=points.dtype)
+        A = tf.cast(self.A, dtype=points.dtype)
+        fval = tf.linalg.matmul(points, A, transpose_b=True)
+        fval = tf.reshape(fval, (*points.shape[:-1], -1))  # Atleast 2D.
+        values = [fval - lb, ub - fval]
+        values = tf.concat(values, axis=-1)
+        return values
+
+    def __repr__(self) -> str:
+        """"""
+        return f"""
+            LinearConstraint({self.A!r}, {self.lb!r}, {self.ub!r}, {self.keep_feasible!r})"
+        """
+
+    def __eq__(self, other: object) -> bool:
+        """
+        :param other: A constraint.
+        :return: Whether the constraint is identical to this one.
+        """
+        if not isinstance(other, LinearConstraint):
+            return False
+        return bool(
+            tf.reduce_all(self.A == other.A)
+            and tf.reduce_all(self.lb == other.lb)
+            and tf.reduce_all(self.ub == other.ub)
+            and tf.reduce_all(self.keep_feasible == other.keep_feasible)
+        )
+
+
+# Constraint type alias.
```

Comment:
If you turn this comment into a docstring it will show up in the API docs (e.g. see the docstring for AcquisitionFunction)

Reply:
Done in b76379c.